omc3/optics_measurements/coupling.py

```diff
@@ -217,19 +217,34 @@ def compensate_rdts_ryoichi():
 
 # ----- Helpers ----- #
 
-# def _take_next(phases: tfs.TfsDataFrame, shift: int = 1):
-#     """
-#     Takes the following BPM for momentum reconstruction by a given shift.
-#
-#     Args:
-#         phases (tfs.TfsDataFrame): Dataframe matrix of phase advances, as calculated in phase.py.
-#         shift (int): ???
-#     """
-#     indices = np.roll(np.arange(phases.to_numpy().shape[0]), shift)
-#     return indices, phases.to_numpy()[np.arange(phases.to_numpy().shape[0]), indices] - 0.25
+def _find_pair(phases: tfs.TfsDataFrame, mode: int = 1):
+    """
+    Does the BPM pairing
+
+    Args:
+        mode (int): 0) find best candidate, >=1) take n-th following BPM
+
+    """
+
+    if mode == 0:
+        return _find_candidate(phases)
+    else:
+        return _take_next(phases, mode)
+
+
+def _take_next(phases: tfs.TfsDataFrame, shift: int = 1):
```

Comment:
Same here for the default value

Reply:
well, here it should be 1 or removed.

0 doesn't make sense and will give a dataframe full of `nan`s

Reply:
Yeah makes sense.